service/internal/accessgroup/accessgroup.go

```diff
@@ -0,0 +1,34 @@
+package accessgroup
+
+type AccessGroup struct {
+	Name string
+}
+
+type AccessGroups struct {
+	WriteAllCustomerGroups   []AccessGroup
+	WriteAllGiantswarmGroups []AccessGroup
+}
+
+func (a *AccessGroups) AddLegacyCustomerAdminGroup(legacyGroupName string) {
+	a.WriteAllCustomerGroups = addLegacyGroupIfMissing(a.WriteAllCustomerGroups, legacyGroupName)
+}
+
+func (a *AccessGroups) AddLegacyGiantswarmAdminGroup(legacyGroupName string) {
+	a.WriteAllGiantswarmGroups = addLegacyGroupIfMissing(a.WriteAllGiantswarmGroups, legacyGroupName)
+}
+
+func (a *AccessGroups) Validate() bool {
+	return len(a.WriteAllCustomerGroups) > 0 && len(a.WriteAllGiantswarmGroups) > 0
```

Comment:
Do we have a write all customer group on all installations?

Reply:
Maybe not. 
There is definitely a different validation mechanism for customer groups, so I removed it from that specific place in the code and only kept validation of GS admin groups.